### PR TITLE
feat(drawer): Make list instance publicly accessible

### LINKED ISSUE #4364: Make List public in MDC Drawer
Make `this.list_` public in `MDCDrawer` in order to programmatically set properties on the `MDCList`, such as `selectedIndex`.

[Example use case on Glitch](https://glitch.com/~material-tabs-to-drawer)

### PR DESCRIPTION
**Pull request by kfranqueiro**
Fixes #4364.

### CODE REVIEW

File: packages/mdc-drawer/component.ts [hunk 1]
```diff
@@ -68,6 +68,10 @@ export class MDCDrawer extends MDCComponent<MDCDismissibleDrawerFoundation> {
   private handleKeydown_!: SpecificEventListener<'keydown'>; // initialized in initialSyncWithDOM()
   private handleTransitionEnd_!: SpecificEventListener<'transitionend'>; // initialized in initialSyncWithDOM()
 
+  get list(): MDCList | undefined {
```
Comment (abhiomkar) [inline packages/mdc-drawer/component.ts]: May not be related to this PR. Looking at the implementation of drawer component, `list_` should not be `undefined`, drawer component should throw error when list is not found instead. Drawer foundation breaks at `focusActiveNavigationItem` adapter method if list is not found.

Comment (kfranqueiro) [reply packages/mdc-drawer/component.ts]: I had wondered about this too, but we have the `list_` member currently declared as `MDCList?` and it looks like we're tolerating it being absent in `destroy`. So maybe there's indeed a separate bug to be had here...

Comment (kfranqueiro) [reply packages/mdc-drawer/component.ts]: Upon further inspection/discussion, it looks like the adapter APIs also tolerate the list not existing, so in the interests of maintaining current behavior I'll keep this PR as-is.

**Comment by tychenjiajun** [comment]
It will be better if I can listen to MDCList:action in the drawer.

**Pull request merged**